--- a/ansible/group_vars/all.yml
+++ b/ansible/group_vars/all.yml
@@ -193,6 +193,13 @@ default_extra_volumes: []
 # Arbitrary unique number from 0..255
 keepalived_virtual_router_id: "51"
 
+#######################
+# Fluentd options
+#######################
+
+# Enable the additional watch timer
+fluentd_enable_watch_timer: "false"
+
 ######################
 # Elasticsearch Options
 #######################
--- a/ansible/roles/common/templates/conf/input/00-global.conf.j2
+++ b/ansible/roles/common/templates/conf/input/00-global.conf.j2
@@ -20,6 +20,6 @@
   keep_time_key true
   time_format %F %T.%L
   ignore_repeated_permission_error true
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
 </source>
 {%- endif %}
--- a/ansible/roles/common/templates/conf/input/02-mariadb.conf.j2
+++ b/ansible/roles/common/templates/conf/input/02-mariadb.conf.j2
@@ -8,7 +8,7 @@
   format multiline
   format_firstline /^(\d{4}-\d{2}-\d{2}|\d{6}) /
   format1 /^(?<Payload>.*)/
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
 </source>
 <source>
   @type tail
@@ -22,5 +22,5 @@
   keep_time_key true
   time_format %y/%m/%d@%T
   ignore_repeated_permission_error true
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
 </source>
--- a/ansible/roles/common/templates/conf/input/03-rabbitmq.conf.j2
+++ b/ansible/roles/common/templates/conf/input/03-rabbitmq.conf.j2
@@ -6,5 +6,5 @@
   format multiline
   format_firstline /^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d{3}/
   format1 /^(?<Timestamp>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d{3}) \[(?<log_level>\w+)\] (?<Payload>.*)/
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
 </source>
--- a/ansible/roles/common/templates/conf/input/04-openstack-wsgi.conf.j2
+++ b/ansible/roles/common/templates/conf/input/04-openstack-wsgi.conf.j2
@@ -5,5 +5,5 @@
   pos_file /var/run/td-agent/kolla-openstack-wsgi.pos
   tag kolla.*
   format /^(?<Payload>.*)$/
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
 </source>
--- a/ansible/roles/common/templates/conf/input/05-libvirt.conf.j2
+++ b/ansible/roles/common/templates/conf/input/05-libvirt.conf.j2
@@ -6,5 +6,5 @@
   format /^(?<Timestamp>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3}\+\d{4}): (?<Pid>\d+): (?<log_level>\S+) : (?<Payload>.*)?$/
   time_key Timestamp
   time_format %F %T.%L%z
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
 </source>
--- a/etc/kolla/globals.yml
+++ b/etc/kolla/globals.yml
@@ -181,6 +181,13 @@
 # where the VIPs of different regions reside on a common subnet.
 #keepalived_virtual_router_id: "51"
 
+#######################
+# Fluentd options
+#######################
+
+# Enable the additional watch timer
+fluentd_enable_watch_timer: "false"
+
 ###################
 # Dimension options
 ###################
