From 5c11c009c669f45751ac5db4033ffcc478609298 Mon Sep 17 00:00:00 2001
From: Christian Berendt <berendt@osism.tech>
Date: Thu, 08 Apr 2021 08:40:23 +0200
Subject: [PATCH] Add fluentd_enable_watch_timer

Change-Id: Ic8ce6fbc3ed8f31d5d090e114b35703532679729
---

diff --git a/ansible/roles/common/defaults/main.yml b/ansible/roles/common/defaults/main.yml
index dad0418..685e4a2 100644
--- a/ansible/roles/common/defaults/main.yml
+++ b/ansible/roles/common/defaults/main.yml
@@ -127,6 +127,9 @@
 # Fluentd
 ####################
 
+# Enable the additional watch timer
+fluentd_enable_watch_timer: "false"
+
 fluentd_input_openstack_services:
   - name: aodh
     enabled: "{{ enable_aodh | bool }}"
diff --git a/ansible/roles/common/templates/conf/input/00-global.conf.j2 b/ansible/roles/common/templates/conf/input/00-global.conf.j2
index af6a4fe..e778662 100644
--- a/ansible/roles/common/templates/conf/input/00-global.conf.j2
+++ b/ansible/roles/common/templates/conf/input/00-global.conf.j2
@@ -14,7 +14,7 @@
   pos_file /var/run/td-agent/kolla-openstack.pos
   tag kolla.*
   ignore_repeated_permission_error true
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
   <parse>
     @type multiline
     format_firstline /^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3} \d+ \S+ \S+ \[(req-\S+ \S+ \S+ \S+ \S+ \S+|-)\]/
diff --git a/ansible/roles/common/templates/conf/input/02-mariadb.conf.j2 b/ansible/roles/common/templates/conf/input/02-mariadb.conf.j2
index f4ed045..23456e3 100644
--- a/ansible/roles/common/templates/conf/input/02-mariadb.conf.j2
+++ b/ansible/roles/common/templates/conf/input/02-mariadb.conf.j2
@@ -5,7 +5,7 @@
   path /var/log/kolla/mariadb/mariadb.log
   pos_file /var/run/td-agent/mariadb.pos
   tag infra.mariadb
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
   <parse>
     @type multiline
     format_firstline /^(\d{4}-\d{2}-\d{2}|\d{6}) /
@@ -18,7 +18,7 @@
   pos_file /var/run/td-agent/mariadb-xinetd.pos
   tag infra.mariadb-xinetd
   ignore_repeated_permission_error true
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
   <parse>
     @type multiline
     format_firstline /^\d{2}/\d{1,2}/\d{1,2}@\d{1,2}:\d{1,2}:\d{1,2}\: (START|EXIT)\: /
diff --git a/ansible/roles/common/templates/conf/input/03-rabbitmq.conf.j2 b/ansible/roles/common/templates/conf/input/03-rabbitmq.conf.j2
index e0e74b1..340e7ee 100644
--- a/ansible/roles/common/templates/conf/input/03-rabbitmq.conf.j2
+++ b/ansible/roles/common/templates/conf/input/03-rabbitmq.conf.j2
@@ -3,7 +3,7 @@
   path /var/log/kolla/rabbitmq/rabbit@{{ ansible_facts.hostname }}.log
   pos_file /var/run/td-agent/rabbit.pos
   tag infra.rabbit
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
   <parse>
     @type multiline
     format_firstline /^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d{3}/
diff --git a/ansible/roles/common/templates/conf/input/04-openstack-wsgi.conf.j2 b/ansible/roles/common/templates/conf/input/04-openstack-wsgi.conf.j2
index f94edf8..eb3fe63 100644
--- a/ansible/roles/common/templates/conf/input/04-openstack-wsgi.conf.j2
+++ b/ansible/roles/common/templates/conf/input/04-openstack-wsgi.conf.j2
@@ -4,7 +4,7 @@
   path /var/log/kolla/*/*-access.log,/var/log/kolla/*/*-error.log,/var/log/kolla/*/*_access.log,/var/log/kolla/*/*_error.log
   pos_file /var/run/td-agent/kolla-openstack-wsgi.pos
   tag kolla.*
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
   <parse>
     @type regexp
     expression /^(?<Payload>.*)$/
diff --git a/ansible/roles/common/templates/conf/input/05-libvirt.conf.j2 b/ansible/roles/common/templates/conf/input/05-libvirt.conf.j2
index 249e52b..365208f 100644
--- a/ansible/roles/common/templates/conf/input/05-libvirt.conf.j2
+++ b/ansible/roles/common/templates/conf/input/05-libvirt.conf.j2
@@ -3,7 +3,7 @@
   path /var/log/kolla/libvirt/libvirtd.log
   pos_file /var/run/td-agent/libvirt.pos
   tag infra.libvirt
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
   <parse>
     @type regexp
     expression /^(?<Timestamp>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3}\+\d{4}): (?<Pid>\d+): (?<log_level>\S+) : (?<Payload>.*)?$/
diff --git a/ansible/roles/common/templates/conf/input/10-openvswitch.conf.j2 b/ansible/roles/common/templates/conf/input/10-openvswitch.conf.j2
index f08272b..db555c3 100644
--- a/ansible/roles/common/templates/conf/input/10-openvswitch.conf.j2
+++ b/ansible/roles/common/templates/conf/input/10-openvswitch.conf.j2
@@ -3,7 +3,7 @@
   path /var/log/kolla/openvswitch/ovs-vswitchd.log
   pos_file /var/run/td-agent/openvswitch.pos
   tag infra.openvswitch
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
   <parse>
     @type multiline
     format_firstline /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}/
@@ -18,7 +18,7 @@
   path /var/log/kolla/openvswitch/ovsdb-server.log
   pos_file /var/run/td-agent/openvswitchdb.pos
   tag infra.openvswitchdb
-  enable_watch_timer false
+  enable_watch_timer {{ fluentd_enable_watch_timer }}
   <parse>
     @type multiline
     format_firstline /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}/
diff --git a/releasenotes/notes/add-parameter-fluentd-enable-watch-timer-08fbe65ba2b549e8.yaml b/releasenotes/notes/add-parameter-fluentd-enable-watch-timer-08fbe65ba2b549e8.yaml
new file mode 100644
index 0000000..d4b13e3
--- /dev/null
+++ b/releasenotes/notes/add-parameter-fluentd-enable-watch-timer-08fbe65ba2b549e8.yaml
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    With the boolean parameter ``fluentd_enable_watch_timer`` it is
+    now possible to enable the additional watch timer of Fluentd.
+    It is disabled by default.
