--- a/ansible/roles/rabbitmq/defaults/main.yml
+++ b/ansible/roles/rabbitmq/defaults/main.yml
@@ -39,3 +39,4 @@ rabbitmq_dimensions: "{{ default_container_dimensions }}"
 rabbitmq_user: "openstack"
 rabbitmq_cluster_name: "openstack"
 rabbitmq_hostname: "{{ ansible_hostname }}"
+rabbitmq_server_additional_erl_args: ""
diff --git a/ansible/roles/rabbitmq/templates/rabbitmq-env.conf.j2 b/ansible/roles/rabbitmq/templates/rabbitmq-env.conf.j2
--- a/ansible/roles/rabbitmq/templates/rabbitmq-env.conf.j2
+++ b/ansible/roles/rabbitmq/templates/rabbitmq-env.conf.j2
@@ -1,6 +1,6 @@
 RABBITMQ_NODENAME=rabbit@{{ ansible_hostname }}
 RABBITMQ_BOOT_MODULE=rabbit_clusterer
-RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS="-pa /usr/lib/rabbitmq/lib/{{ rabbitmq_version }}"
+RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS="-pa /usr/lib/rabbitmq/lib/{{ rabbitmq_version }} {{ rabbitmq_server_additional_erl_args }}"
 
 RABBITMQ_LOG_BASE=/var/log/kolla/{{ project_name }}
 
