diff --git a/ansible/roles/elasticsearch/templates/elasticsearch.json.j2 b/ansible/roles/elasticsearch/templates/elasticsearch.json.j2
index f34d2ea3b..317ae5658 100644
--- a/ansible/roles/elasticsearch/templates/elasticsearch.json.j2
+++ b/ansible/roles/elasticsearch/templates/elasticsearch.json.j2
@@ -3,7 +3,7 @@
     "config_files": [
         {
             "source": "{{ container_config_directory }}/elasticsearch.yml",
-            "dest": "/usr/share/elasticsearch/config/elasticsearch.yml",
+            "dest": "/etc/elasticsearch/elasticsearch.yml",
             "owner": "elasticsearch",
             "perm": "0600"
         }
diff --git a/ansible/roles/elasticsearch/templates/elasticsearch.yml.j2 b/ansible/roles/elasticsearch/templates/elasticsearch.yml.j2
index b85391424..007764ae6 100644
--- a/ansible/roles/elasticsearch/templates/elasticsearch.yml.j2
+++ b/ansible/roles/elasticsearch/templates/elasticsearch.yml.j2
@@ -14,9 +14,7 @@ http.port: {{ elasticsearch_port }}
 gateway.expected_nodes: {{ num_nodes }}
 gateway.recover_after_time: "5m"
 gateway.recover_after_nodes: {{ recover_after_nodes }}
-path.conf: "/etc/elasticsearch"
 path.data: "/var/lib/elasticsearch/data"
 path.logs: "/var/log/kolla/elasticsearch"
-path.scripts: "/etc/elasticsearch/scripts"
 indices.fielddata.cache.size: 40%
 action.auto_create_index: "true"
