--- a/ansible/roles/kibana/tasks/config.yml
+++ b/ansible/roles/kibana/tasks/config.yml
@@ -33,6 +33,17 @@
   notify:
     - Restart kibana container
 
+- name: Copying over Logtrail configuration file
+  template:
+    src: "logtrail.json.j2"
+    dest: "{{ node_config_directory }}/{{ item.key }}/logtrail.json"
+  when:
+    - inventory_hostname in groups[item.value.group]
+    - item.value.enabled | bool
+  with_dict: "{{ kibana_services }}"
+  notify:
+    - Restart kibana container
+
 - name: Check kibana containers
   kolla_docker:
     action: "compare_container"
diff --git a/ansible/roles/kibana/templates/kibana.json.j2 b/ansible/roles/kibana/templates/kibana.json.j2
index 3ce5c7f8..7186540e 100644
--- a/ansible/roles/kibana/templates/kibana.json.j2
+++ b/ansible/roles/kibana/templates/kibana.json.j2
@@ -6,6 +6,12 @@
             "dest": "/opt/kibana/config/kibana.yml",
             "owner": "kibana",
             "perm": "0640"
+        },
+        {
+            "source": "{{ container_config_directory }}/logtrail.json",
+            "dest": "/usr/share/kibana/plugins/logtrail/logtrail.json",
+            "owner": "kibana",
+            "perm": "0640"
         }
     ],
     "permissions": [
diff --git a/ansible/roles/kibana/templates/logtrail.json.j2 b/ansible/roles/kibana/templates/logtrail.json.j2
new file mode 100644
index 00000000..2ac1df9f
--- /dev/null
+++ b/ansible/roles/kibana/templates/logtrail.json.j2
@@ -0,0 +1,31 @@
+{
+  "index_patterns" : [
+    {      
+      "es": {
+        "default_index": "*",
+        "allow_url_parameter": false       
+      },
+      "tail_interval_in_seconds": 10,
+      "es_index_time_offset_in_seconds": 0,
+      "display_timezone": "local",
+      "display_timestamp_format": "MMM DD HH:mm:ss",
+      "max_buckets": 500,
+      "nested_objects" : false,
+      "default_time_range_in_days" : 0,
+      "max_hosts": 100,
+      "max_events_to_keep_in_viewer": 5000,
+      "fields" : {
+        "mapping" : {
+            "timestamp" : "@timestamp",
+            "display_timestamp" : "@timestamp",
+            "hostname" : "Hostname",
+            "program": "programname",
+            "message": "message"
+        },
+{% raw %}
+        "message_format": "{{hostname}}: {{message}}"
+{% endraw %}
+      }
+    }  
+  ]
+}
