--- a/ansible/roles/rabbitmq/defaults/main.yml
+++ b/ansible/roles/rabbitmq/defaults/main.yml
@@ -38,4 +38,5 @@ rabbitmq_dimensions: "{{ default_container_dimensions }}"
 ####################
 rabbitmq_user: "openstack"
 rabbitmq_cluster_name: "openstack"
+rabbitmq_cluster_version: 1
 rabbitmq_hostname: "{{ ansible_hostname }}"
--- a/ansible/roles/rabbitmq/templates/rabbitmq-clusterer.config.j2
+++ b/ansible/roles/rabbitmq/templates/rabbitmq-clusterer.config.j2
@@ -1,5 +1,5 @@
 [
-  {version, 1},
+  {version, {{ rabbitmq_cluster_version }}},
   {nodes, [
   {% for host in groups[role_rabbitmq_groups] %}
     {'rabbit@{{ hostvars[host]['ansible_hostname'] }}', disc}
