From f9e4c4418e8e425d1adea530c20a1be4b2d65c58 Mon Sep 17 00:00:00 2001
From: Christian Berendt <berendt@osism.tech>
Date: Fri, 15 Mar 2024 16:12:31 +0100
Subject: [PATCH] octavia: add missing bool to the enable_octavia_jobboard parameter

Change-Id: I9304f3546b20c0406e195163dccb1433fe802204
---

diff --git a/ansible/roles/octavia/templates/octavia.conf.j2 b/ansible/roles/octavia/templates/octavia.conf.j2
index 6f046bb..7d7efd8 100644
--- a/ansible/roles/octavia/templates/octavia.conf.j2
+++ b/ansible/roles/octavia/templates/octavia.conf.j2
@@ -154,7 +154,7 @@
 region_name = {{ openstack_region_name }}
 endpoint_type = internal
 ca_certificates_file = {{ openstack_cacert }}
-{% if enable_octavia_jobboard %}
+{% if enable_octavia_jobboard | bool %}
 
 [task_flow]
 persistence_connection = mysql+pymysql://{{ octavia_persistence_database_user }}:{{ octavia_persistence_database_password }}@{{ octavia_persistence_database_address }}/{{ octavia_persistence_database_name }}
