--- a/ansible/roles/horizon/templates/_9998-kolla-settings.py.j2
+++ b/ansible/roles/horizon/templates/_9998-kolla-settings.py.j2
@@ -18,7 +18,16 @@ DATABASES = {
 }
 {% elif groups['memcached'] | length > 0 and not horizon_backend_database | bool %}
 SESSION_ENGINE = 'django.contrib.sessions.backends.cache'
+{% endif %}
+
+{% if groups['memcached'] | length > 0 %}
 CACHES['default']['LOCATION'] = [{% for host in groups['memcached'] %}'{{ 'api' | kolla_address(host) | put_address_in_context('url') }}:{{ memcached_port }}'{% if not loop.last %},{% endif %}{% endfor %}]
+CACHES['default']['OPTIONS'] =  {
+    "no_delay": True,
+    "ignore_exc": True,
+    "max_pool_size": 4,
+    "use_pooling": True,
+}
 {% endif %}

 {% if kolla_enable_tls_external | bool or kolla_enable_tls_internal | bool %}
